### DISCUSSION TITLE
C++ 20 Support

### DISCUSSION THREAD

**Discussion by sajalsan** [eclipse-iceoryx/iceoryx#2077] [General]
In my understanding currently IceOryx currently can be built using C++ 17.
Also there are restrictions that all apps using Road must be built using same compiler and version.
Is there a plan to support C++20?

I would like to use Modules/Jthread from C++20 in my program

**Comment by elBoberido** [comment]
@sajalsan currently we are planing the transition from C++14 to C++17 as minimum C++ version. It will take some time until we move to C++20.

While you cannot use modules with iceoryx yet, you should be able to compile your code with C++20 and use jthreads. Just use set the respective C++ version in your cmake target, e.g.
```cmake
set_target_properties(lib_or_binary_target
                      PROPERTIES
                      CXX_STANDARD_REQUIRED ON
                      CXX_STANDARD 20
)
```

**Reply by sajalsan** [reply]
thank you so much, I will try.
